Source/WebCore/css/MediaQueryMatcher.cpp

```diff
@@ -39,15 +39,14 @@
 namespace WebCore {
 
 MediaQueryMatcher::MediaQueryMatcher(Document& document)
-    : m_document(document)
+    : ContextDestructionObserver(&document)
 {
 }
 
 MediaQueryMatcher::~MediaQueryMatcher() = default;
 
 void MediaQueryMatcher::documentDestroyed()
```

Comment:
This should override `contextDestroyed()` from ContextDestructionObserver instead. then we would no longer need this code in Document.cpp:
```
    if (RefPtr mediaQueryMatcher = m_mediaQueryMatcher)
        mediaQueryMatcher->documentDestroyed();
```